pkg/server/commands/list_objects_test.go

```diff
@@ -22,3 +31,124 @@ func TestNewListObjectsQuery(t *testing.T) {
 		require.Error(t, err)
 	})
 }
+
+func TestCheckDispatchCount(t *testing.T) {
+	ds := memory.New()
+	ctx := storage.ContextWithRelationshipTupleReader(context.Background(), ds)
+
+	tests := []struct {
+		name                  string
+		model                 string
+		tuples                []*openfgav1.TupleKey
+		objectType            string
+		relation              string
+		user                  string
+		expectedDispatchCount uint32
+	}{
+		{
+			name: "test_direct_relation",
+			model: `model
+schema 1.1
+
+type user
+
+type folder
+  	relations
+		define viewer: [user] 
+`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("folder:C", "viewer", "user:jon"),
+				tuple.NewTupleKey("folder:B", "viewer", "user:jon"),
+				tuple.NewTupleKey("folder:A", "viewer", "user:jon"),
+			},
+			objectType:            "folder",
+			relation:              "viewer",
+			user:                  "user:jon",
+			expectedDispatchCount: 3,
+		},
+		{
+			name: "test_union_relation",
+			model: `model
+schema 1.1
+
+type user
+
+type folder
+  	relations
+		define editor: [user]
+		define viewer: [user] or editor 
+`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("folder:C", "editor", "user:jon"),
+				tuple.NewTupleKey("folder:B", "viewer", "user:jon"),
+				tuple.NewTupleKey("folder:A", "viewer", "user:jon"),
+			},
+			objectType:            "folder",
+			relation:              "viewer",
+			user:                  "user:jon",
+			expectedDispatchCount: 4,
+		},
+		{
+			name: "test_intersection_relation",
+			model: `model
+schema 1.1
+
+type user
+
+type folder
+  	relations
+		define editor: [user]
+		define can_delete: [user] and editor 
+`,
+			tuples: []*openfgav1.TupleKey{
+				tuple.NewTupleKey("folder:C", "can_delete", "user:jon"),
+				tuple.NewTupleKey("folder:B", "viewer", "user:jon"),
+				tuple.NewTupleKey("folder:A", "viewer", "user:jon"),
+			},
+			objectType:            "folder",
+			relation:              "can_delete",
+			user:                  "user:jon",
+			expectedDispatchCount: 2,
+		},
+	}
+
```

Comment:
Maybe add a couple zero cases to ensure accuracy. 
```suggestion
{
			name: "no_tuples",
			model: `model
schema 1.1

type user

type folder
  	relations
		define editor: [user]
		define can_delete: [user] and editor 
`,
			tuples:                []*openfgav1.TupleKey{},
			objectType:            "folder",
			relation:              "can_delete",
			user:                  "user:jon",
			expectedDispatchCount: 0,
		},
```